test/e2e/proxy_test.go

```diff
@@ -1084,3 +1078,153 @@ func testProxyServiceLifeCycle(ipFamily *corev1.IPFamily, ingressIPs []string, d
 		}
 	}
 }
+
+// TestProxyLoadBalancerModeDSR creates a LoadBalancer Service and verifies both external and internal clients accessing
+// its LoadBalancer IPs work as expected.
+// Client IP should always be preserved regardless of whether the traffic is externally or internally originated.
+// Session affinity should take effect when it's enabled.
+func TestProxyLoadBalancerModeDSR(t *testing.T) {
+	skipIfHasWindowsNodes(t)
+	skipIfNumNodesLessThan(t, 3)
+	skipIfAntreaIPAMTest(t)
+	skipIfProxyDisabled(t)
+
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+	skipIfProxyAllDisabled(t, data)
+	skipIfEncapModeIsNot(t, data, config.TrafficEncapModeEncap)
+
+	ingressNode := controlPlaneNodeName()
+	backendNode1 := workerNodeName(1)
+	backendNode2 := workerNodeName(2)
+
+	internalClient := "internal-client"
+	if err := NewPodBuilder(internalClient, data.testNamespace, toolboxImage).OnNode(ingressNode).Create(data); err != nil {
+		t.Fatalf("Failed to create internal client: %v", err)
```

Comment:
you may want to use `require.NoError` instead of `t.Fatal` since you use it in other places, for less verbosity